### DISCUSSION TITLE
Query a one-to-many relationship, select the many

### DISCUSSION THREAD

**Discussion by tehfailsafe** [supabase/supabase#7958] [Questions] [answered]
Hello, first day using supabase and have been stuck on this simple query. Most examples show the one-to-many relationship by illustrating fetching the single entity from the many item. (For example, showing the artist of a album when an artist has many items)

I am trying to select a list of top 10 artists and return all the albums they have created in a single query. 

When I try the to select 'albums (name)' on an artists 'from' it returns null. The UI for the artist table shows the albums relationship as null, but if I click the little button next to it I see a dialog with the correct albums. Under the albums table I do see the correct artist for each album.

I could run a nested query for each artist, but wouldn't that hurt performance? I've been used to grapql so maybe I'm spoiled...

**Comment by silentworks** [comment, marked as answer]
You can do this with a single query
```js
await supabase.from('artist').select(`*, album(name)`)
```
But you need to make sure your relationship inside the database is correctly setup.

**Reply by tehfailsafe** [reply]
That's what I'm trying, but I get null for the result. I setup a reference to artist in the album table, and a corresponding ref to albums in the artist table. The album has a value there (the id of the artist) but what should the value be for the artist? It's currently null since I want to have a one-to-many relationship, it doesn't make sense that I should need to define it in the artist table as an array, what would the point be then of the reference?

Also I tinkered with the graphql api version and I can get the results I want using query > artist > albumsCollection > name, even with the null value in the artist table for albums field. I am doing this in SSR though so I would prefer a simpler rest api version if possible.

![image](https://user-images.githubusercontent.com/1388454/182007685-816f230a-f59d-45c3-b1d5-e002b85ec0dd.png)

**Reply by GaryAustin1** [reply]
Normally you don't have both foreign keys to each other (especially for a one to many relationship).   Just drop the artist table albums column.

That same API call should then work.   Well except you may need to flush PostgREST's schema cache since you are changing an existing relationship...  Run this in the SQL editor after you make the change:  NOTIFY pgrst, 'reload schema'

Edit:  You probably will have to put "Albums" in quotes in the API call as you are using caps.  Unless you are using Prisma, you are better off not using caps in Postgres.

**Reply by tehfailsafe** [reply]
Perfect, that worked. A bit counter intuitive coming from ages ago in rails where you needed the pointer on the Artist table, but I'm happy it works!

**Reply by tehfailsafe** [reply]
Ah, thanks for the tip on casing, old habits die hard I suppose.

**Reply by steve-chavez** [reply]
> Edit: You probably will have to put "Albums" in quotes in the API call as you are using caps

FYI: That's unnecessary when querying through the API, only when using [reserved characters](https://postgrest.org/en/stable/api.html#reserved-characters)(`,`,`.`,`:`, `()`) you need to double quote. Uppercase characters or [unicode characters](https://postgrest.org/en/stable/api.html#unicode-support) should work fine.
